db/fetcher/fetcher.go

```diff
@@ -13,27 +13,38 @@ package fetcher
 import (
 	"bytes"
 	"context"
+	"fmt"
+	"strings"
 
+	"github.com/bits-and-blooms/bitset"
 	dsq "github.com/ipfs/go-datastore/query"
 
 	"github.com/sourcenetwork/defradb/client"
 	"github.com/sourcenetwork/defradb/core"
 	"github.com/sourcenetwork/defradb/datastore"
 	"github.com/sourcenetwork/defradb/datastore/iterable"
 	"github.com/sourcenetwork/defradb/db/base"
+	"github.com/sourcenetwork/defradb/planner/mapper"
+	"github.com/sourcenetwork/defradb/request/graphql/parser"
 )
 
 // Fetcher is the interface for collecting documents from the underlying data store.
 // It handles all the key/value scanning, aggregation, and document encoding.
 type Fetcher interface {
-	Init(col *client.CollectionDescription, fields []*client.FieldDescription, reverse bool, showDeleted bool) error
+	Init(col *client.CollectionDescription, fields []client.FieldDescription, filter *mapper.Filter, docmapper *core.DocumentMapping, reverse bool, showDeleted bool) error
 	Start(ctx context.Context, txn datastore.Txn, spans core.Spans) error
 	FetchNext(ctx context.Context) (*encodedDocument, error)
 	FetchNextDecoded(ctx context.Context) (*client.Document, error)
 	FetchNextDoc(ctx context.Context, mapping *core.DocumentMapping) ([]byte, core.Doc, error)
 	Close() error
 }
 
+// KeyValue is a KV store response containing the resulting core.Key and byte array value.
+type KeyValue struct {
```

Comment:
question: Does this need to be public? Asking because it doesn't seem to be use outside of this file.

Reply:
Nevermind. `DocumentFetcher.KV()` returns this type.

Reply:
It doesn't need to be public. The `NextKV`, `NextKey`, and `ProcessKV` public methods all don't need to exist, theyre old methods mostly for debugging, they can be removed, so KeyValue can be made private.

Reply:
Implemented now